imports/wasi_snapshot_preview1/fs.go

```diff
@@ -804,10 +806,9 @@ func maxDirents(entries []fs.DirEntry, bufLen uint32) (bufused, direntCount, tru
 			// when the next entry is larger than bytes remaining.
 			bufused = bufLen
 
-			// When the next entry is truncated, only write the dirent.
-			if lenRemaining < direntSize {
-				truncatedEntryLen = direntSize - lenRemaining
-			}
+			// We do have enough space to write the header, this value will be
+			// passed on to writeDirents to only write the header for this entry.
+			writeTruncatedEntry = true
```

Comment:
I changed this to a boolean because we don't really need to keep track of the remaining size. 
I removed the condition because we always want to write away the header if there is room enough for it. If there is no room for the header we already break out in the statement above.